src/operations/query/r4.js

```diff
@@ -101,142 +101,82 @@ class R4SearchQueryCreator {
+                // ignore this unrecognized arg
+                continue;
+            }
 
-                    if (args[`${queryParameter}:missing`]) {
-                        filterByMissing({
-                            args, queryParameter, propertyObj, columns
-                        }).forEach(q => and_segments.push(negation ? {$nor: [q]} : q));
-                    } else if (args[`${queryParameter}:contains`]) {
-                        filterByContains({
-                            propertyObj, queryParameter, args, columns
-                        }).forEach(q => and_segments.push(negation ? {$nor: [q]} : q));
-                    } else if (args[`${queryParameter}:above`] && args[`${queryParameter}:below`]) {
-                        filterByAboveAndBelow({
-                            propertyObj, args, queryParameter, columns
-                        }).forEach(q => and_segments.push(negation ? {$nor: [q]} : q));
-                    } else if (args[`${queryParameter}:above`]) {
-                        filterByAbove({
-                            propertyObj, args, queryParameter, columns
-                        }).forEach(q => and_segments.push(negation ? {$nor: [q]} : q));
-                    } else if (args[`${queryParameter}:below`]) {
-                        filterByBelow({
-                            propertyObj, args, queryParameter, columns
-                        }).forEach(q => and_segments.push(negation ? {$nor: [q]} : q));
-                    } else if (args[`${queryParameter}:text`]) {
-                        filterByPartialText({
-                            args, queryParameter, propertyObj, columns,
-                        }).forEach(q => and_segments.push(q));
-                    } else if (args[`${queryParameter}:not:text`] || args[`${queryParameter}:text:not`]) {
-                        filterByPartialText({
-                            args, queryParameter, propertyObj, columns,
-                        }).forEach(q => and_segments.push({$nor: [q]}));
-                    }
+            // set type of field in propertyObj
+            propertyObj.fieldType = this.fhirTypesManager.getTypeForField(
+                {
+                    resourceType,
+                    field: propertyObj.field
+                }
+            );
+            /**
+             * @type {string | string[]}
+             */
+            let queryParameterValue = args[argName];
+            queryParameterValue = convertGraphQLParameters(
+                queryParameterValue,
+                args,
+                queryParameter
+            );
+
+            if (queryParameterValue) {
+
+                let {columns, andSegments} = this.getColumnsAndSegmentsForParameterType({
+                    resourceType, queryParameter, queryParameterValue, propertyObj
+                });
+
+                // replace andSegments according to modifiers
+                if (modifiers.includes('missing')) {
```

Comment:
note that in the old version, the datatype logic (now in getColumnsAndSegmentsForParameterType as called on line 143) was executed exclusive of this modifier logic. now the datatype stuff is evaluated and then the modifier stuff is evaluated